MoneyCompass/Transaction/SavingGoalsView.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  SavingGoalsView.swift
+//  MoneyCompass
+//
+//  Created by MacBook Pro on 10/12/23.
+//
+
+import SwiftUI
+
+struct CircularProgressView: View {
+    let progress: Double
+    var body: some View {
+        ZStack {
+            ZStack {
+                Circle()
+                    .stroke(
+                        Color.gray.opacity(0.3),
+                        lineWidth: 6
+                    )
+                Circle()
+                    .trim(from: 0, to: progress)
+                    .stroke(
+                        Color.accentColor,
+                        style: StrokeStyle(
+                            lineWidth: 6,
+                            lineCap: .round
+                        )
+                    )
+                    .rotationEffect(.degrees(-90))
+                    .animation(.easeOut, value: progress)
+                
+            }
+            Text("\(progress * 100, specifier: "%.0f")%")
+                .fontWeight(.semibold)
+        }
+        .frame(width: .infinity, height: .infinity)
+        .padding(3)
+    }
+}
+
+struct Goals: Identifiable {
+    var id : String {
+        self.name
+    }
+    var name: String
+    var price: Double
+    var progress: Double
+}
+
+struct SavingGoalsView: View {
+    @State var progress: Double = 1
+    @State var goals: [Goals] = [
```

Comment:
Try to fetch saving goals from BudgetGoals  (core data) instead of showing dummy data